src/net/sourceforge/plantuml/ugraphic/ImageParameter.java

```diff
@@ -84,15 +90,21 @@ public ImageParameter(ColorMapper colorMapper, boolean useHandwritten, Animation
 		this.lengthAdjust = LengthAdjust.defaultValue();
 	}
 
-	public ImageParameter(ISkinParam skinParam, Animation animation, double dpiFactor, String metadata,
-			String warningOrError, ClockwiseTopRightBottomLeft margins, HColor backcolor) {
+	public ImageParameter(TitledDiagram diagram, ColorMapper colorMapper, boolean useHandwritten, Animation animation, double dpiFactor,
```

Comment:
This new constructor will hopefully be  short lived.

I think some similar refactorings could end up with a simple constructor like `ImageParameter(TitledDiagram diagram)`.  The logic for backcolor, dpiFactor etc could all be moved into `ImageParameter` so they behave the same for more diagram types.